app/sfr-search-api/lib/v3Edition.js

```diff
@@ -0,0 +1,140 @@
+   */
+  constructor(app, editionIdentifier) {
+    this.app = app
+    this.editionID = editionIdentifier
+    this.edition = null
+    this.logger = this.app.logger
+    this.dbConn = new DBConnection(this.logger, app.dbClient)
+  }
+
+  /**
+   * Method to load and format an object containing edition metadata from the database
+   *
+   * @returns {object} Constructed edition record that can be sent to the end user
+   */
+  async loadEdition() {
+    const editions = await this.getEditions()
+    this.edition = editions[0]
+
+    if (this.edition === undefined) {
+      throw new NotFoundError(`No edition found matching identifier ${this.editionID}`)
+    }
+
+    // Transform source metadata into output format
+    await this.parseEdition()
+
+    // Transform source metadata from child records
+    Helpers.parseAgents(this.edition, 'instances')
+    Helpers.parseLinks(this.edition, 'instances')
+    Helpers.parseDates(this.edition, 'instances')
+
+    return this.edition
+  }
+
+  /**
+   * Handler function for retrieving a set of editions specified by their internal
+   * postgres row ids through the DBConnection class
+   *
+   * @param {array} editionIds Array of Row IDs to the editions table
+   *
+   * @returns {array} Array of edition objects from the database
+   */
+  getEditions() {
+    return this.dbConn.loadEditions([this.editionID], 1)
+  }
+
+  /**
+   * Handler function for retrieving a set of instances related to the current
+   * edition record by the editions PostgreSQL row ID through the DBConnection class
+   *
+   * @returns {array} Array of instance objects from the database
+   */
+  getInstances() {
+    return this.dbConn.getEditionInstances(this.editionID)
+  }
+
+  /**
+   * Method for performing some basic transformations of source metadata for presentation
+   * to end users. This includes parsing dates, selecting the most appropriate title for
+   * the edition and sorting the component instances.
+   */
+  async parseEdition() {
+    this.edition.instances = await this.getInstances()
+
+    // Select the publication year from the publication date range
+    if (this.edition.publication_date) {
+      this.edition.publication_date = this.edition.publication_date.match(/^\[([0-9]+)/)[1]
+    }
+
+    // Count all unique instance titles, sorting them into an object
+    const titleCounts = this.edition.instances
+      .map(i => i.title)
+      .reduce((titles, title) => {
+        const count = titles[title] || 0
+        // eslint-disable-next-line no-param-reassign
+        titles[title] = count + 1
+        return titles
+      }, {})
+
+    // Select the most common instance title as the edition title
+    this.edition.title = Object.keys(titleCounts)
+      .sort((a, b) => titleCounts[b] - titleCounts[a])[0]
+
+    // Perform a custom sort on the instance array
+    this.sortInstances()
+
+    // Remove items and covers from the edition; these are displayed on instances
+    delete this.edition.items
```

Comment:
I think we go through this over and over again but ... you're gonna have to explain to me the difference between items and instances again sorry orz

Reply:
Oh I see below, so you have `instance.items`.  got it.

Reply:
yeah, in the database items (and their links) are always related to instances. For the work detail we pull out one item and add it to the edition for the card.

This reuses some of that work detail code, so we just have to remove that item from the edition since for the work detail all items are attached to instances